### PR TITLE
qa: Run all tests even if wallet is not compiled

### PR DESCRIPTION
**Pull request by maflcko**
Currently the test_runner would exit if the wallet was not compiled into the Bitcoin Core executable. However, a lot of the tests run without the wallet just fine and there is no need to globally require the wallet to run the tests.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Travis run for reference: https://travis-ci.org/bitcoin/bitcoin/jobs/426386820#L2937

**Comment by lucash-dev** [comment]
This PR includes a commit that's also present in #14179 (fae497d). What's the relationship between the two PRs?

**Comment by Sjors** [comment]
Concept ACK. It's useful to (occasionally) run the test suite without wallet (and other stuff) compiled. 

Quite a few tests are skipped this way, but that can be improved later. That might require pre-generated transactions (only when `--disable-wallet`), or the ability for a node to generate transactions without having full wallet functionality.

How will this interact with #10102 (multi process), which has one binary with and one without a wallet? cc @ryanofsky. I imagine a common test configuration involves one wallet binary to generate transactions as well as one node binary.

I tested on macOS (only the `--disable-wallet` variant).

```
WARNING! There is already a bitcoind process running on this system. Tests may fail unexpectedly due to resource contention!

wallet_hd.py skipped
```

Can you add one `WARNING!` at the top for each disabled feature, so it's more clear what caused the `skipped` lines?

**Comment by MarcoFalke** [comment]
> Quite a few tests are skipped this way, but that can be improved later.

Agree that this can be improved later. Skipping quite a few tests is better than skipping all tests.

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -247,6 +259,14 @@ def setup_nodes(self):
         self.add_nodes(self.num_nodes, extra_args)
         self.start_nodes()
 
+    def import_deterministic_coinbase_privkeys(self):
+        for n in self.nodes:
+            wallet_enabled = 'Wallet' in [line[3:-3] for line in n.help().splitlines() if line.startswith('==')]
+            if not wallet_enabled:
+                continue
+
+            n.importprivkey(n.get_deterministic_priv_key()[1])
```
Comment (Empact) [inline test/functional/test_framework/test_framework.py]: nit: could invert the conditional / drop the continue

File: test/functional/test_framework/test_node.py [hunk 2]
```diff
@@ -97,6 +97,22 @@ def __init__(self, i, datadir, *, rpchost, timewait, bitcoind, bitcoin_cli, mock
...
+            ('msX6jQXvxiNhx3Q62PKeLPrhrqZQdSimTg', 'cUxsWyKyZ9MAQTaAhUQWJmBbSvHMwSmuv59KgxQV7oZQU3PXN3KE'),
+            ('mnonCMyH9TmAsSj3M59DsbH8H63U3RKoFP', 'cTrh7dkEAeJd6b3MRX9bZK8eRmNqVCMH3LSUkE3dSFDyzjU38QxK'),
+            ('mqJupas8Dt2uestQDvV2NH3RU8uZh2dqQR', 'cVuKKa7gbehEQvVq717hYcbE9Dqmq7KEBKqWgWrYBa2CKKrhtRim'),
+            ('msYac7Rvd5ywm6pEmkjyxhbCDKqWsVeYws', 'cQDCBuKcjanpXDpCqacNSjYfxeQj8G6CAtH1Dsk3cXyqLNC4RPuh'),
+            ('n2rnuUnwLgXqf9kk2kjvVm8R5BZK1yxQBi', 'cQakmfPSLSqKHyMFGwAqKHgWUiofJCagVGhiB4KCainaeCSxeyYq'),
+            ('myzuPxRwsf3vvGzEuzPfK9Nf2RfwauwYe6', 'cQMpDLJwA8DBe9NcQbdoSb1BhmFxVjWD5gRyrLZCtpuF9Zi3a9RK'),
+            ('mumwTaMtbxEPUswmLBBN3vM9oGRtGBrys8', 'cSXmRKXVcoouhNNVpcNKFfxsTsToY5pvB9DVsFksF1ENunTzRKsy'),
+            ('mpV7aGShMkJCZgbW7F6iZgrvuPHjZjH9qg', 'cSoXt6tm3pqy43UMabY6eUTmR3eSUYFtB2iNQDGgb3VUnRsQys2k'),
+        ]
+        return PRIV_KEYS[self.index]
```
Comment (Empact) [inline test/functional/test_framework/test_node.py]: Raise an error if index exceeds PRIV_KEYS?

**Comment by laanwj** [comment]
> Agree that this can be improved later. Skipping quite a few tests is better than skipping all tests.

:+1:

**Comment by Sjors** [comment]
The [x86_64 Linux, No wallet](https://github.com/bitcoin/bitcoin/blob/master/.travis.yml#L120) Travis host [already runs this](https://travis-ci.org/bitcoin/bitcoin/jobs/426908084):
schermafbeelding 2018-09-11 om 15 47 39

**Comment by jnewbery** [comment]
Concept ACK

**Comment by laanwj** [comment]
utACK

**Comment by ryanofsky** [comment]
Note: Some code review comments on this PR can be found in #14179 instead of here, since this PR was originally based on #14179, but I think got accidentally merged first.

**Pull request merged**